codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -357,42 +357,31 @@ let rec pruneAndMergeNextRegistered = (self: t, ~isMerged=false) => {
 Updates node at given id with given values and checks to see if it can be merged into its next register.
 Returns Error if the node with given id cannot be found (unexpected)
 
-fetchedEvents are ordered earliest to latest (as they are returned from the worker)
+newItems are ordered earliest to latest (as they are returned from the worker)
 */
-let update = (self: t, ~id, ~latestFetchedBlock, ~fetchedEvents, ~currentBlockHeight): result<
+let update = (self: t, ~id, ~latestFetchedBlock, ~newItems, ~currentBlockHeight): result<
   t,
   exn,
 > => {
   let isFetchingAtHead =
     currentBlockHeight <= latestFetchedBlock.blockNumber ? true : self.isFetchingAtHead
   self
-  ->updateInternal(~id, ~latestFetchedBlock, ~newFetchedEvents=fetchedEvents, ~isFetchingAtHead)
+  ->updateInternal(
+    ~id,
+    ~latestFetchedBlock,
+    ~reversedNewItems=newItems->Array.reverse,
```

Comment:
I still don't see the optimization here? It's just reversing earlier rather than later?

Reply:
There's no. I just kept the code. Explained my misunderstanding here https://github.com/enviodev/hyperindex/pull/272#discussion_r1805037788